source/common/config/grpc_mux_impl.h

```diff
@@ -119,6 +120,18 @@ class GrpcMuxImpl : public GrpcMux,
   struct ApiState {
     bool paused() const { return pauses_ > 0; }
 
+    void ensureTtlManagerCreated(Event::Dispatcher& dispatcher,
+                                 std::function<void(const std::vector<std::string>&)> callback) {
+      // TODO(snowp): This pattern is a bit gross, but it keeps ApiState default allocatable which
```

Comment:
Can't this be some internal anonymous namespace function?

Reply:
The main reason I did this was because ApiState is currently always being allocated inline via operator[], so if we introduce a non-trivial ctor then we'd break operator[] and would have to replace all of those call sites with calls to `find()` (or some helper). 

It's totally doable but I wanted to avoid that churn since the PR is already big enough, though I would be happy to do it now if that's desired

Reply:
Sure, I think maybe move to an anonymous helper that factors out some of this pattern. Thanks.